### ISSUE TITLE
Keep medias sending order.

### ISSUE THREAD

**Issue by ylecollen** [vector-im/vector-android#258]
Currently, if the user selects several images to upload, the medias are uploaded in parallel.
Asap, their content is uploaded, the dedicated message is sent.

But, the order might be wrong. For example, the first media is a large image and the second is a small one.

The second media content will be uploaded quickly and the dedicated message will be sent before the first media.

**Comment by Soundtoxin** [comment]
Yes, this is killing me. The order isn't kept in riot-web either. Not sure if there's a similar issue opened there yet. I'd like it to keep track of the order I selected files in, OR give me an option to re-order before sending, OR at the least do alphabetical order. If I post some screenshots I took of a video, the filenames have timestamps and they *always* send out of order. This is one of a handful of issues with Riot that I can't get over and that routinely make my day less pleasant.
